modules/top-level/output.nix

```
@@ -172,9 +172,10 @@ in
   config =
     let
       # Optionally byte compile lua library
+      inherit (import ./plugins/byte-compile-lua-lib.nix { inherit lib pkgs; }) byteCompileLuaPackages;
```

Ideally:

```suggestion
      inherit (pkgs.callPackages ./plugins/byte-compile-lua-lib.nix { }) byteCompileLuaPackages;
```

If it doesn't involve too much out-of-scope refactoring.

EDIT: I'm not sure how well that'd work with accessing `lib.nixvim.builders`, since the `lib` available via `callPackages` may not be the same lib used by nixvim's module eval.

---

I prefer callPackage myself, but the latest refactor in #3138 used many imports, so thought that this is a recommended style.

> EDIT: I'm not sure how well that'd work with accessing lib.nixvim.builders, since the lib available via callPackages may not be the same lib used by nixvim's module eval.

callPackage by default sets arguments to the ones from pkgs, so I think the lib would be from pkgs.lib. If you doesn't populate it with your lib, I think it won't work without callPackageWith. I'll check.

---

> but the latest refactor in #3138 used many imports, so thought that this is a recommended style.

Generally we're moving towards using `callPackage`, `callPackages`, or custom callers like our `callTest` or `callLib` implementations of `callPackageWith`. Sometimes this gets overlooked even in new code, though :grin: 

> callPackage [and callPackages] by default sets arguments to the ones from pkgs, so I think the lib would be from pkgs.lib.

Correct

> If nixvim doesn't populate it with your nixvim's lib, I think it won't work without callPackageWith.

Kinda.

You don't necessarily need to set custom `autoArgs` via `callPackagesWith`, because you can also set specify args explicitly. E.g.:

```nix
pkgs.callPackages ./foo { bar = someOverride; }

# AKA
lib.callPackagesWith pkgs ./foo { bar = someOverride; }
```

The question is whether its worth it at that point.

---

I've added a separate commit on top that changes all usages of import in top-level modules to `callPackage .. (inherit lib;)`. See if it's better. If not, I'll drop the commit. I didn't change `inherit (import ./utils.nix lib)` though.